sklearn/linear_model/coordinate_descent.py

```diff
@@ -470,7 +473,9 @@ def enet_path(X, y, l1_ratio=0.5, eps=1e-3, n_alphas=100, alphas=None,
         if dual_gap_ > eps_:
             warnings.warn('Objective did not converge.' +
                           ' You might want' +
-                          ' to increase the number of iterations',
+                          ' to increase the number of iterations.' +
+                          ' Fitting float32 data with small alpha,' +
```

Comment:
This warning should definitely be decoupled with the existing warning.

You can move the other warning to the start of fit, when the dtype of X is float32 and alpha is less than 1e-8 instead of confusing users with a unrelated message when the algorithm has not converged.